### PR TITLE
Split ktor-server-test-host module

### PR DESCRIPTION
**Pull request by rsinukov**
**Motivation**
Extract tests so that `:apiCheck` doesn't trigger when new tests are added
https://youtrack.jetbrains.com/issue/KTOR-1075

### CODE REVIEW

**Comment by cy6erGn0m** [comment]
Unfortunately, it contains a testing API (functions like `withTestApplication` and the test host implementation). We probably need to split the module into two

**Comment by rsinukov** [comment]
@cy6erGn0m thanks, I split the module into two

File: gradle/compatibility.gradle [hunk 1]
```diff
@@ -11,7 +11,8 @@ apiValidation {
         'ktor-client',
         'ktor-client-features',
         'ktor-client-benchmarks',
-        'ktor-server-benchmarks'
+        'ktor-server-benchmarks',
+        'ktor-server-test-host-suites',
```
Comment (e5l) [inline gradle/compatibility.gradle:15]: maybe we can name it `ktor-server-test-sutes`?

File: ktor-server/ktor-server-test-host-suites/build.gradle [hunk 2]
```diff
@@ -0,0 +1,11 @@
+description = ''
+
+kotlin.sourceSets.jvmMain.dependencies {
+    implementation project(':ktor-server:ktor-server-test-host')
+
+    implementation "org.jetbrains.kotlinx:kotlinx-coroutines-debug:$coroutines_version"
+}
+
+dependencies {
+    jvmTestApi project(path: ':ktor-server:ktor-server-core', configuration: 'testOutput')
```
Comment (e5l) [inline ktor-server/ktor-server-test-host-suites/build.gradle:10]: Can be done in a modern way:
```
kotlin.sourceSets {
    jvmMain { 
    }
    jvmTest { 
    }
}
```

**Review by e5l — CHANGES_REQUESTED** [review]
LGTM, except minor problems

**Review by e5l — APPROVED** [review]

File: ktor-server/ktor-server-test-suites/build.gradle [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        dependencies {
+            implementation project(':ktor-server:ktor-server-test-host')
+
+            implementation "org.jetbrains.kotlinx:kotlinx-coroutines-debug:$coroutines_version"
+        }
+    }
+}
+
+dependencies {
+    jvmTestApi project(path: ':ktor-server:ktor-server-core', configuration: 'testOutput')
```
Comment (e5l) [inline ktor-server/ktor-server-test-suites/build.gradle:14]: same here

**Review by e5l — CHANGES_REQUESTED** [review]

**Review by e5l — APPROVED** [review]
Good job, lgtm

**Pull request merged**